core/src/test/java/org/infinispan/scattered/statetransfer/CoordinatorStopTest.java

```diff
@@ -0,0 +1,168 @@
+      cb.clustering().hash().consistentHashFactory(chf);
+
+      addClusterEnabledCacheManager(cb);
+      addClusterEnabledCacheManager(cb);
+      // If the updated topologies from old coord come when it's no longer in the view these are ignored.
+      // Therefore we have to delay the view.
+      viewLatch = new CountDownLatch(1);
+      GlobalConfigurationBuilder gcb = new GlobalConfigurationBuilder();
+      gcb.transport().transport(new DelayedViewJGroupsTransport(viewLatch));
+      addClusterEnabledCacheManager(gcb, cb);
+
+      assertTrue(cache(0).getCacheManager().isCoordinator());
+      // start other caches
+      cache(1);
+      cache(2);
+      waitForClusterToForm();
+
+      // segment 0 will be moved to cache(2). Since we've lost coord cache(1) now -> 0 and cache(2) -> 1
+      chf.setOwnerIndexes(new int[][] { {1}, {0}, {1} });
+   }
+
+   // Reproducer for ISPN-9128
+   public void testCoordinatorLeaves() throws InterruptedException, ExecutionException {
+      MagicKey key = new MagicKey(cache(0));
+      cache(1).put(key, "value");
+      int stableTopologyId = cache(1).getAdvancedCache().getDistributionManager().getCacheTopology().getTopologyId();
+      BlockingLocalTopologyManager ltm2 = BlockingLocalTopologyManager.replaceTopologyManager(manager(2), cache(2).getName());
+      ControlledRpcManager rpcManager2 = ControlledRpcManager.replaceRpcManager(cache(2));
+      rpcManager2.excludeCommands(StateResponseCommand.class);
+
+      log.infof("Stopping coordinator %s, last stable topology is %d", manager(0), stableTopologyId);
+      manager(0).stop();
+
+      log.infof("Current members are: %s", cache(1).getCacheManager().getMembers());
+      assertTrue(cache(1).getCacheManager().isCoordinator());
+
+      // topology + 1 is the one that just omits the leaving node
+      BlockedTopology t1 = ltm2.expectTopologyUpdate(CacheTopology.Phase.NO_REBALANCE, stableTopologyId + 1);
+      assertEquals(2, t1.getCacheTopology().getActualMembers().size());
+      assertEquals(null, t1.getCacheTopology().getPendingCH());
+      assertOwners(t1, true, 0);
+      assertOwners(t1, true, 1, address(1));
+      assertOwners(t1, true, 2, address(2));
+      t1.unblock();
+
+      // topology + 2 has TRANSITORY phase and all segments have an owner in pendingCH
+      BlockedTopology t2 = ltm2.expectTopologyUpdate(CacheTopology.Phase.TRANSITORY, stableTopologyId + 2);
+      assertEquals(2, t2.getCacheTopology().getActualMembers().size());
+      assertNotNull(t2.getCacheTopology().getPendingCH());
+      assertOwners(t2, false, 0, address(2));
+      assertOwners(t2, false, 1, address(1));
+      assertOwners(t2, false, 2, address(2));
+      t2.unblock();
+
+      // Let the rebalance begin
+      rpcManager2.expectCommand(StateRequestCommand.class,
+            request -> assertEquals(StateRequestCommand.Type.CONFIRM_REVOKED_SEGMENTS, request.getType()))
+            .send().receiveAll();
```

Comment:
It's weird that I wrote `expectCommand` to only allow access to the command in the consumer, but I'm not going to argue with myself now :)